src/PyNvCodec/src/PyNvDecoder.cpp

```diff
@@ -624,6 +624,98 @@ bool PyNvDecoder::DecodeSurface(DecodeContext& ctx)
   }
 }
 
+auto copy_nv12_to_nv12Planar = [](shared_ptr<Surface> src, shared_ptr<Surface> dst)->bool{
```

Comment:
could be `shared_ptr&` since it is not necessary to increment the refcount just for this function scope

Reply:
those variables could be static or in anonymous namespace

Reply:
where is this lambda used? Couldn't your superior NV12 type replace the old one to avoid confusion?